- `01_Intro.Rmd` - Introduction and course logistics
- `02_Genomics.Rmd` - Overview of sequencing technologies

## `hw` - homework material

## `lab` - in-class lab material

## `read` - reading assignment

### General

- [Milestones in Genomic Sequencing](https://www.nature.com/immersive/d42859-020-00099-0/index.html) by Nature, 2000-2021 period, interactive infographics

- Claussnitzer, Melina, Judy H. Cho, Rory Collins, Nancy J. Cox, Emmanouil T. Dermitzakis, Matthew E. Hurles, Sekar Kathiresan, et al. “[A Brief History of Human Disease Genetics](https://doi.org/10.1038/s41586-019-1879-7).” Nature, (January 2020) - Timeline and milestones in genomics development focusing on disease genetics.

- Hagen, J. B. “[The Origins of Bioinformatics](https://doi.org/10.1038/35042090).” Nature Reviews. Genetics, (01 December 2000)

- [The (near) complete sequence of a human genome](https://genomeinformatics.github.io/CHM13v1/) - efforts for completing human genome assembly

- Mardis, Elaine R. “[DNA Sequencing Technologies: 2006-2016](https://doi.org/10.1038/nprot.2016.182).” Nature Protocols, (February 2017) - DNA sequencing technologies introduction, references.

- Heather, James M., and Benjamin Chain. “[The Sequence of Sequencers: The History of Sequencing DNA](https://doi.org/10.1016/j.ygeno.2015.11.003).” Genomics, (January 2016) - Review of sequencing technologies, from pre-Sanger to current PacBio, Oxford Nanopore, Ion Torrent.

- Goodwin, Sara, John D. McPherson, and W. Richard McCombie. “[Coming of Age: Ten Years of next-Generation Sequencing Technologies](https://doi.org/10.1038/nrg.2016.49).” Nature Reviews. Genetics, (17 2016) - From microarrays to long read sequencing, single-cell sequencing. Technology review. Table with all technologies and costs. Figures.

- Green, Eric D., Mark S. Guyer, Eric D. Green, Mark S. Guyer, Teri A. Manolio, and Jane L. Peterson. “[Charting a Course for Genomic Medicine from Base Pairs to Bedside](https://doi.org/10.1038/nature09764).” Nature, (February 10, 2011) - Perspective on genomics development, technologies. Figure 1 - pictorial roadmap

- Mardis, Elaine R. “[Next-Generation DNA Sequencing Methods](https://doi.org/10.1146/annurev.genom.9.081307.164359).” Annual Review of Genomics and Human Genetics, (2008) - Sequencing technologies review. DNA-/RNA-/ChIP-seq. Figures.

- E. S. Lander et al., “[Initial Sequencing and Analysis of the Human Genome](https://doi.org/10.1038/35057062),” Nature, (February 15, 2001) - Genome sequencing landmark paper.

<!--
- Cordaux, Richard, and Mark A. Batzer. “[The Impact of Retrotransposons on Human Genome Evolution](https://doi.org/10.1038/nrg2640).” Nature Reviews. Genetics, (October 2009)

- Rothberg, Jonathan M, and John H Leamon. “The Development and Impact of 454 Sequencing.” Nature Biotechnology 26, no. 10 (October 2008): 1117–24. doi:10.1038/nbt1485. - 454 sequencing, history of sequencing development, pyrosequencing. Other technologies - Box 1, Table 2
-->

### Genomics resources

- [SequencEnG](http://education.knoweng.org/sequenceng/index.html) - An interactive learning resource for next-generation sequencing (NGS) techniques. Each technology had the corresponding reference and infographics.

- [Links and references to many resources](https://github.com/crazyhottommy/getting-started-with-genomics-tools-and-resources) by Ming Tang

- [Bioinformatics-Resources](https://github.com/YTLogos/Bioinformatics-Resources) -  A curated list of resources for learning bioinformatics.

- [biotools](https://github.com/jdidion/biotools) - A massive collection of references on the topics of bioinformatics, sequencing technologies, programming, machine learning, and more.

- [Computational genomics with R](https://compgenomr.github.io/book/) book by Altuna Akalin. [GitHub repo](https://github.com/compgenomr/book)

### Misc

- [https://www.illumina.com/documents/products/techspotlights/techspotlight_sequencing.pdf](https://www.illumina.com/documents/products/techspotlights/techspotlight_sequencing.pdf) - read about Illumina technology, mostly figures

- Video. The Genomic Landscape circa 2016 - Eric Green. Overview of genomics and medicine. [https://www.youtube.com/watch?v=mhD3-_5Ee-A](https://www.youtube.com/watch?v=mhD3-_5Ee-A)

- Video. Any talks by Elaine Mardis. [https://www.youtube.com/results?search_query=elaine+mardis]( https://www.youtube.com/results?search_query=elaine+mardis)

# ToDo



 Q1 - From 1 (least) to 10 (highest), rate your knowledge of/familiarity with each of the following:
 - NCBI
 - UCSC genome browser
 - ENSEMBL genome browser
 - R and RStudio
 - Galaxy
 - Gene Expression Omnibus (GEO)
 - Short Read Archive (SRA)
 - The Cancer Genome Atlas (TCGA) project
 - GenomeSpace
 - Jupyter Notebooks

  Q2 - From 1 (no knowledge) to 10 (expert), rate your expertise on each of the following:
  - Genetics
  - Molecular Biology
  - Cell biology
  - Probability
  - Statistics
  - Regression Analysis
  - Computational Biology/Bioinformatics
  - Using cloud computing resources
  - Data analysis with R
  - Matlab

   Q3 - Which of the following terms could you define without the use of Google or another search engine (select all that apply):

   - Chromatin modification
   - RNA transcription
   - Gene editing
   - Protein translation
   - Burrows-Wheeler Transform
   - Sequencing Alignment
   - Expectation-Maximization
   - Bayesian Inference

 Q4 - From 1 (not interested) to 5 (very interested), rate your interest in learning to work with the following data types. Please select N/A if you are not familiar with the data type.
 - Exome seq
 - Chip-seq
 - Poly-A sequencing
 - ATAC-seq
 - Histone-seq
 - RNA-seq
 - Whole Genome seq
 - Mass Spectrometry
 - Eukaryotic Organisms

  Q5 - Which are you more interested in data generated from:
  - Human
  - A highly studied model organism (mouse, zebrafish, C. elegans)
  - Rare/unusual/non-model organism
  - A mix of all
  - No preference

 Q10 - Do you have genomic data you might bring to the course?
 - Yes/no
Q11 - Please describe data you may bring to the course:
 Q6 - Do you have access to a computing cluster that students will be able to login to and perform calculations?
 - Yes/no
 

# `misc`



