<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Cancer Bioinformatics course</title>
    <meta charset="utf-8" />
    <meta name="author" content="Mikhail Dozmorov" />
    <script src="libs/header-attrs-2.6/header-attrs.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
    <link rel="stylesheet" href="xaringan-my.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Cancer Bioinformatics course
## Introduction
### Mikhail Dozmorov
### Virginia Commonwealth University
### 01-25-2021

---




## Computational Biology

&lt;center&gt;&lt;img src="img/computational_biology.png" height="300px" /&gt;&lt;/center&gt;

&amp;nbsp;

&amp;nbsp;

Markowetz, Florian. "[All Biology Is Computational Biology](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2002050)" _PLoS Biology_ March 9, 2017

---
## Welcome!

The primary goal of the course is to provide theory and practice of computational genomics, and empower you to conduct independent genomic analyses

- We will study the leading computational approaches for analyzing genomes starting from raw sequencing data

- The course will focus on cancer genomics and human medical applications, but the techniques will be broadly applicable

- The topics will include genomic technologies, alignment algorithms, genome assembly, gene- and miRNA expression, methylation and epigenomic analysis, single nucleotide polymorphism and copy number variant analysis, metagenomics, single-cell analysis, 3D genomics

---
## Logistics

- Course Webpage: https://bios691-cancer-bioinformatics.netlify.app/
    - Lecture notes, exercises R code, references will be posted there

- Course announcements, assignments, homework submission and grading, class video recordings
    - Blackboard, https://blackboard.vcu.edu

- Class Hours: Monday/Wednesday, 9:00 am to 10:20 am
    - Attendance is encouraged but not required. You can watch lectures later, but be mindful of homework deadlines

- Office Hours:  by appointment

---
## Virtual format

- Let's maintain a welcoming and supportive environment
    
- Video on is encouraged - it increases _your_ attention

- Ask as many questions during the class as necessary - this is what live attendance is for
    - Use Chat functionality
    - If anyone knows answers, post them there

---
## Slack

For the off-class questions/announcements, we'll use [Slack](https://slack.com/). See [Slack 101](https://slack.com/resources/slack-101) if you are new to it

- I will send invites using your VCU e-mail addresses - use them to join the [#bios691_cancer_bioinformatics channel](https://dozmorovlab.slack.com/archives/C01J4NKL1B7)

- Omit formalities, but be respectful and polite

- Questions and answers posted there will be available for everyone - answer them if you can, or I'll address them

- Direct messages are possible - use them to ask private questions or schedule a meeting

- I'll try to answer your questions asap, typically within 24 hours

---
## Prerequisites

- No formal course requirements, but basic knowledge of the following will help
    - Basic statistics knowledge: descriptive statistics, estimators, (linear) modeling (e.g., BIOS 544 or 554 courses)
    - Basic programming skills in R, familiarity with command line (e.g., BIOS 524 course)

- Hardware
    - A laptop, Mac or Linux OSs are recommended

---
## Computational resources

- We will be working within **Linux environment**
    - If you are on Windows, you may install [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10). Alternatively, [Cygwin](https://www.cygwin.com/). It should enable a Linux command line environment (bash) on your local machine
    - Alternatively, contact Helen Wang (huwang at vcu.edu) to get an account on our computer cluster, `merlot.bis.vcu.edu`
    - VCU Biostatistics cluster information, https://wiki.vcu.edu/display/biosit/Home

- We will program in [R](https://www.r-project.org/) and use [RStudio](https://rstudio.com/products/rstudio/download/)
    - Install R and RStudio on your machine

---
## Class Resources

- Primary Texts: We will be studying research papers and notes

- Lecture slides will contain links. Make good use of them, explore the references on your own!

- Other Resources: Google, SEQanswers, Biostars, StackOverflow

---
## Grading Policies

- Homework assignments practicing methods and tools learned in class: 50% of the total grade
    - Due two weeks after the assignment

- Final Project developing a genomics analysis: 50%
    - Pre-proposal and due dates are to be discussed

- Collaborations are allowed, no more than two per group
    - Duplicate or nearly identical homeworks will receive a score of zero

---
## Course evaluation

- At the end of the course, you will be asked to evaluate it

- Assess what you learned during the course

- Take notes on what you like in the course, what you want to be improved

- Your evaluation is anonymous


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9",
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
