---
title: "Alignment algorithms"
# linktitle: "1. Introduction to RStudio and Images"
date: "2021-02-17"
# class_date: "2021-01-25"
output:
  blogdown::html_page:
    toc: false
menu:
  reading:
    parent: Reading
    weight: 7
type: docs
# weight: 1 
# pdf: 
# thumb: 
editor_options: 
  chunk_output_type: console
---

<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>


<div id="online-demos-and-tools" class="section level2">
<h2>Online demos and tools</h2>
<ul>
<li><p><a href="http://teacheng.illinois.edu/#nw">TeachEnG</a> - educational tool for aslgnment algorithms, plylogenetic tree. - Kim, Minji, Yeonsung Kim, Lei Qian, and Jun S. Song. “<a href="https://doi.org/10.1093/bioinformatics/btx447">TeachEnG: A Teaching Engine for Genomics</a>.” Bioinformatics, October 15, 2017</p></li>
<li><p><a href="http://experiments.mostafa.io/public/needleman-wunsch/">Needleman-Wunsch algorithm demo</a></p></li>
<li><p><a href="https://github.com/life4/textdistance">textdistance</a> - compute distance between sequences. 30+ algorithms, pure python implementation, common interface, optional external libs usage</p></li>
</ul>
</div>
<div id="alignment-algorithms" class="section level2">
<h2>Alignment algorithms</h2>
<ul>
<li><p>Needleman, S. B., and C. D. Wunsch. “<a href="https://doi.org/10.1016/0022-2836(70)90057-4">A General Method Applicable to the Search for Similarities in the Amino Acid Sequence of Two Proteins</a>.” Journal of Molecular Biology, March 1970</p></li>
<li><p>Smith, T. F., and M. S. Waterman. “<a href="https://doi.org/10.1016/0022-2836(81)90087-5">Identification of Common Molecular Subsequences</a>.” Journal of Molecular Biology, March 25, 1981</p></li>
<li><p>Burrows, Michael, and David J Wheeler. “<a href="http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.3.8069">A Block-Sorting Lossless Data Compression Algorithm</a>,” 1994. - BWT paper</p></li>
<li><p>Ferragina, Paolo, and Giovanni Manzini. “<a href="https://dl.acm.org/citation.cfm?id=796543">Opportunistic Data Structures with Applications</a>.” In Foundations of Computer Science, IEEE, 2000 - FM index paper</p></li>
<li><p>Li, Heng, and Richard Durbin. “<a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and Accurate Short Read Alignment with Burrows-Wheeler Transform</a>.” Bioinformatics, July 15, 2009 - BWA first paper</p></li>
<li><p>Li, Heng, and Richard Durbin. “<a href="https://doi.org/10.1093/bioinformatics/btp698">Fast and Accurate Long-Read Alignment with Burrows-Wheeler Transform</a>.” Bioinformatics, March 1, 2010 - BWA second paper</p></li>
<li><p>Li, Heng. “<a href="https://arxiv.org/abs/1303.3997">Aligning Sequence Reads, Clone Sequences and Assembly Contigs with BWA-MEM</a>.” ArXiv, 2013. - BWA-MEM (maximal exact matches). Automatically select end-to-end or gapped alignment.</p></li>
<li><p>Langmead, Ben, and Steven L Salzberg. “<a href="https://doi.org/10.1038/nmeth.1923">Fast Gapped-Read Alignment with Bowtie 2</a>.” Nature Methods, March 4, 2012 - Bowtie2 gapped alignment</p></li>
<li><p>Dobin, Alexander, Carrie A. Davis, Felix Schlesinger, Jorg Drenkow, Chris Zaleski, Sonali Jha, Philippe Batut, Mark Chaisson, and Thomas R. Gingeras. “<a href="https://doi.org/10.1093/bioinformatics/bts635">STAR: Ultrafast Universal RNA-Seq Aligner</a>.” Bioinformatics, January 1, 2013 - STAR gapped aligner. Algorithm, testing on simulated dataset.</p></li>
<li><p>Kim, Daehwan, Ben Langmead, and Steven L. Salzberg. “<a href="https://doi.org/10.1038/nmeth.3317">HISAT: A Fast Spliced Aligner with Low Memory Requirements</a>.” Nature Methods, April 2015 - HISAT aligner. Two indexes - one large with many small</p></li>
<li><p>Fonseca, Nuno A., Johan Rung, Alvis Brazma, and John C. Marioni. “<a href="https://doi.org/10.1093/bioinformatics/bts605">Tools for Mapping High-Throughput Sequencing Data</a>.” Bioinformatics, December 15, 2012 - Sequencing mappers, lists and characteristics of DNA, RNA, bisulfite aligners.</p></li>
</ul>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<ul>
<li><p>Nagarajan, Niranjan, and Mihai Pop. “<a href="https://doi.org/10.1038/nrg3367">Sequence Assembly Demystified</a>.” Nature Reviews. Genetics, March 2013 - Gentle introduction into genome assembly. Technologies. Box2: Greedy, overlap-layout-consensus, De Bruijn. Problems</p></li>
<li><p>Pevzner, P. A., H. Tang, and M. S. Waterman. “<a href="https://doi.org/10.1073/pnas.171285098">An Eulerian Path Approach to DNA Fragment Assembly</a>.” Proceedings of the National Academy of Sciences, August 14, 2001 - First de Bruijn graph for genome assembly paper. Idea of breaking reads into fragments. Typical approach reads are vertices connected by edges if they overlap. Hamiltonian path problem - visit each vertex exactly once, NP-complete. de Bruijn graph - overlapping fragments are edges, and the problem is Eulerian path - visit each edge once. Error-correction algorithm.</p></li>
<li><p>Compeau, Phillip E C, Pavel A Pevzner, and Glenn Tesler. “<a href="https://doi.org/10.1038/nbt.2023">How to Apply de Bruijn Graphs to Genome Assembly</a>.” Nature Biotechnology, November 8, 2011</p></li>
<li><p>Chaisson, Mark J. P., Richard K. Wilson, and Evan E. Eichler. “<a href="https://doi.org/10.1038/nrg3933">Genetic Variation and the de Novo Assembly of Human Genomes</a>.” Nature Reviews Genetics, October 7, 2015 - Genome assembling strategies, problems. OLC, De Bruijn, string graphs. Types of gaps.</p></li>
<li><p>Miller, Jason R., Sergey Koren, and Granger Sutton. “<a href="https://doi.org/10.1016/j.ygeno.2010.03.001">Assembly Algorithms for Next-Generation Sequencing Data</a>.” Genomics, June 2010 - Assembly tools for overlap/layout/consensus and the de Bruijn graph approaches. de Bruin graph Issues with genome assembly, potential solutions.</p></li>
<li><p>String Graph Assembler. Simpson, J. T., and R. Durbin. “<a href="http://www.genome.org/cgi/doi/10.1101/gr.126953.111">Efficient de Novo Assembly of Large Genomes Using Compressed Data Structures</a>.” Genome Research, March 1, 2012 - SGA - String Graph Assembler. From an FM-index. Velvet, ABySS, SOAPdenovo de Bruijn graph assemblers. BWA and FM explanation</p></li>
<li><p>Koren, Sergey, and Adam M. Phillippy. “<a href="https://doi.org/10.1016/j.mib.2014.11.014">One Chromosome, One Contig: Complete Microbial Genomes from Long-Read Sequencing and Assembly</a>.” Current Opinion in Microbiology, February 2015 - Genome assembly overview focusing on long reads. Repeats (global and local) are problematic. Details on technologies: PacBio RS, Illumina’s Moleculo, ONT MinION. Assembling approaches: OLC, hierarchical hybrid (long reads correction using another technology) and non-hybrid (self long reads alignment-correction). Assembly augmentation: gap filling, scaffolding, read threading. Table 1 - long read assembly tools and descriptions.</p></li>
</ul>
<div id="videos" class="section level3">
<h3>Videos</h3>
<ul>
<li><p><a href="https://www.youtube.com/playlist?list=PL2mpR0RYFQsBiCWVJSvVAO3OJ2t7DzoHA">Algorithms for DNA Sequencing</a> - 55 short videos by Ben Langmead</p>
<ul>
<li><a href="https://github.com/BenLangmead/ads1-slides.git">Ben Langmead’s Slides for Algorithms for DNA Sequencing Coursera class, GitHub repo</a></li>
<li><a href="http://www.langmead-lab.org/teaching-materials/">Ben Langmead’s teaching material with links to videos, lecture notes, python notebooks</a></li>
</ul></li>
<li><p><a href="https://www.youtube.com/watch?v=lJzybEXmIj0&amp;list=PLUl4u3cNGP63uK-oWiLgO7LLJV6ZCWXac&amp;index=1">MIT 7.91J Foundations of Computational and Systems</a> - full lecture videos (approx 1h 20m), by Christopher Burge, David Gifford, Ernest Fraenkel</p></li>
<li><p>Vector illustrations of BWT based searching on small strings, <a href="https://github.com/ekg/drawbwt" class="uri">https://github.com/ekg/drawbwt</a></p>
<ul>
<li><a href="https://alexbowe.com/fm-index/">FM-Indexes and Backwards Search</a></li>
</ul></li>
</ul>
</div>
</div>
