---
title: "RNA-seq"
# linktitle: "1. Introduction to RStudio and Images"
date: "2021-03-03"
# class_date: "2021-01-25"
output:
  blogdown::html_page:
    toc: false
menu:
  reading:
    parent: Reading
    weight: 9
type: docs
# weight: 1 
# pdf: 
# thumb: 
editor_options: 
  chunk_output_type: console
---

<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>


<ul>
<li><p>Lowe, Rohan, Neil Shirley, Mark Bleackley, Stephen Dolan, and Thomas Shafee. “<a href="https://doi.org/10.1371/journal.pcbi.1005457">Transcriptomics Technologies</a>.” PLoS Computational Biology, (May 2017)</p></li>
<li><p>Conesa, Ana, Pedro Madrigal, Sonia Tarazona, David Gomez-Cabrero, Alejandra Cervera, Andrew McPherson, Michał Wojciech Szcześniak, et al. “<a href="https://doi.org/10.1186/s13059-016-0881-8">A Survey of Best Practices for RNA-Seq Data Analysis</a>.” Genome Biology, (December 2016) - RNA-seq analysis roadmap, QC. Differential detection. TPM. Tools for alternative splicing detection and visualization. small RNA analysis. single cell. Integrative analysis, with methylation.</p></li>
<li><p>Garber, Manuel, Manfred G. Grabherr, Mitchell Guttman, and Cole Trapnell. “<a href="https://doi.org/10.1038/nmeth.1613">Computational Methods for Transcriptome Annotation and Quantification Using RNA-Seq</a>.” Nature Methods, (June 2011) - RNA-seq alignment and quantification. Table of tools. Transcriptome reconstruction. Alternative splicing</p></li>
<li><p>Wang, Zhong, Mark Gerstein, and Michael Snyder. “<a href="https://doi.org/10.1038/nrg2484">RNA-Seq: A Revolutionary Tool for Transcriptomics</a>.” Nature Reviews. Genetics, (January 2009) - RNA-seq review.</p></li>
<li><p>Williams, Alexander G., Sean Thomas, Stacia K. Wyman, and Alisha K. Holloway. “<a href="(https://doi.org/10.1002/0471142905.hg1113s83)">RNA-Seq Data: Challenges in and Recommendations for Experimental Design and Analysis: RNA-Seq Data: Experimental Design and Analysis</a>.” In Current Protocols in Human Genetics, John Wiley &amp; Sons, Inc., 2014. - RNA-seq basics, tools, simulations</p></li>
<li><p>Altman, Naomi, and Martin Krzywinski. “<a href="https://doi.org/10.1038/nmeth.3224">Points of Significance: Sources of Variation</a>.” Nature Methods, (December 30, 2014)</p></li>
<li><p>Martin, Jeffrey A., and Zhong Wang. “<a href="https://doi.org/10.1038/nrg3068">Next-Generation Transcriptome Assembly</a>.” Nature Reviews. Genetics, (September 7, 2011) - Transcriptome assembly. Sequencing technologies overview. Reference-based and de novo assembly, combined approach idea. Splice graph, De Bruijn graph.</p></li>
<li><p>Marioni, John C., Christopher E. Mason, Shrikant M. Mane, Matthew Stephens, and Yoav Gilad. “<a href="https://doi.org/10.1101/gr.079558.108">RNA-Seq: An Assessment of Technical Reproducibility and Comparison with Gene Expression Arrays</a>.” Genome Research, (September 2008) - Illumina sequencing - microarray comparison. Good agreement. Assessing lane effect with hypergeometric distribution. Likelihood ratio test for differential expression. Chi-squared goodness-of-fit test.</p></li>
<li><p>Peixoto, Lucia, Davide Risso, Shane G. Poplawski, Mathieu E. Wimmer, Terence P. Speed, Marcelo A. Wood, and Ted Abel. “<a href="https://doi.org/10.1093/nar/gkv736">How Data Analysis Affects Power, Reproducibility and Biological Insight of RNA-Seq Studies in Complex Datasets</a>.” Nucleic Acids Research, (September 18, 2015) - The importance of RNA-seq normalization and batch effect removal. RUVseq increases power, but the choice of the number of latent variables is important. <a href="https://github.com/drisso/peixoto2015_tutorial">Tutorial: Steps in RNA-seq data processing, normalization, exploratory data analysis</a></p></li>
<li><p>Wang, Eric T., Rickard Sandberg, Shujun Luo, Irina Khrebtukova, Lu Zhang, Christine Mayr, Stephen F. Kingsmore, Gary P. Schroth, and Christopher B. Burge. “<a href="https://doi.org/10.1038/nature07509">Alternative Isoform Regulation in Human Tissue Transcriptomes</a>.” Nature, (November 27, 2008) - Alternative splicing comparison between tissues. ~94% of genes are alternatively transcribed. Variation in alternative splicing is much more between tissues than between individuals.</p></li>
<li><p>Park, Eddie, Zhicheng Pan, Zijun Zhang, Lan Lin, and Yi Xing. “<a href="https://doi.org/10.1016/j.ajhg.2017.11.002">The Expanding Landscape of Alternative Splicing Variation in Human Populations</a>.” The American Journal of Human Genetics, (January 2018) - Alternative splicing, detailed overview</p></li>
</ul>
<div id="statistics" class="section level3">
<h3>Statistics</h3>
<ul>
<li><p>Pachter, Lior. “<a href="https://arxiv.org/abs/1104.3889">Models for Transcript Quantification from RNA-Seq</a>.” ArXiv, 2011. - RNA-seq quantification statistics, expectation-maximization algorithm</p></li>
<li><p>Robinson, Mark D., and Alicia Oshlack. “<a href="https://doi.org/10.1186/gb-2010-11-3-r25">A Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data</a>.” Genome Biology, (2010) - TMM normalization method. Problems with library scaling normalization. Well-written intuitive motivating example. MA plot, trimming outliers, weighted (inverse of the variance) M average after discarding 30% of M outliers and lowest 5% of A values.</p></li>
<li><p>“<a href="https://liorpachter.wordpress.com/2017/08/02/how-not-to-perform-a-differential-expression-analysis-or-science/">How not to perform a differential expression analysis (or science)</a>” blog post by Lior Pachter, about Salmon-kallisto similarities and differences, general references</p></li>
<li><p>Robinson, Mark D., and Gordon K. Smyth. “<a href="https://doi.org/10.1093/biostatistics/kxm030">Small-Sample Estimation of Negative Binomial Dispersion, with Applications to SAGE Data</a>.” Biostatistics (Oxford, England), (April 2008) - Negative Binomial distribution instead of Poisson. Previous models: binomial, Poisson.</p></li>
<li><p>Lun, Aaron T. L., and Gordon K. Smyth. “<a href="https://doi.org/10.1515/sagmb-2017-0010">No Counts, No Variance: Allowing for Loss of Degrees of Freedom When Assessing Biological Variability from RNA-Seq Data</a>.” Statistical Applications in Genetics and Molecular Biology, (April 25, 2017) - Negative impact of genes with zero counts on GLM framework for RNA-seq differential expression analysis. Overdispersion, GLM, quasi-likelihood F-test, adjusting degrees of freedom for zero-count genes.</p></li>
<li><p>Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. “<a href="https://doi.org/10.1186/gb-2014-15-2-r29">Voom: Precision Weights Unlock Linear Model Analysis Tools for RNA-Seq Read Counts</a>.” Genome Biology, (2014) - voom paper</p></li>
<li><p>Love, Michael I, Wolfgang Huber, and Simon Anders. “<a href="https://doi.org/10.1186/s13059-014-0550-8">Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2</a>.” Genome Biology, (December 2014) - DESeq2 paper. Problems with fold-change ranking of genes - proposed solution using shrinkage of FCs. Generalized linear model using Negative Binomial distribution. Borrowing information - genes of similar avarage expression have similar dispersion. rlog-transformation. <a href="https://www.ncbi.nlm.nih.gov/pubmed/20979621">The original DESeq publication</a></p></li>
<li><p>Witten, Daniela M. “<a href="https://doi.org/10.1214/11-AOAS493">Classification and Clustering of Sequencing Data Using a Poisson Model</a>.” The Annals of Applied Statistics, (December 2011) - RNA-seq modeling with Poisson distribution. samples X genes matrix. Derivation of Poisson, negative binomial, using Poisson for linear discriminant analysis and clustering (Poisson dissimilarity).</p></li>
<li><p>Patro, Rob, Geet Duggal, Michael I Love, Rafael A Irizarry, and Carl Kingsford. “<a href="https://doi.org/10.1038/nmeth.4197">Salmon Provides Fast and Bias-Aware Quantification of Transcript Expression</a>.” Nature Methods, (March 6, 2017) - Salmon paper. Pseudo-alignment, or using precomputed alignment to tramscriptome. Dual-phase statistical inference procedure and sample-specific bias models that account for sequence-specific, fragment, GC content, and positional biases. Comparison with kallisto and sailfish. Tests on simulated (Polyester, RSEM-sim) and real (GEUVADIS, SEQC) data. Detailed Methods description. <a href="https://github.com/COMBINE-lab/Salmon">COMBINE-lab/Salmon GitHub repo</a></p></li>
<li><p>Jiang, Hui, and Wing Hung Wong. “<a href="https://doi.org/10.1093/bioinformatics/btp113">Statistical Inferences for Isoform Expression in RNA-Seq</a>.” Bioinformatics, (April 15, 2009) - Alternative splicing statistics - Poisson modeling. Problem - most reads are shared by more than one isoform. How to quantify isoform expression from exon counts. Detailed statistical derivations</p></li>
<li><p>Young, Matthew D., Matthew J. Wakefield, Gordon K. Smyth, and Alicia Oshlack. “<a href="https://doi.org/10.1186/gb-2010-11-2-r14">Gene Ontology Analysis for RNA-Seq: Accounting for Selection Bias</a>.” Genome Biology, (2010) - Gene set enrichment analysis accounting for length and expression of transcripts. Instead of random sampling, use of the Wallenius non-central hypergeometric distribution to account for biased sampling. <a href="https://bioconductor.org/packages/goseq/">GOseq R package</a></p></li>
<li><p>Law, Charity W., Kathleen Zeglinski, Xueyi Dong, Monther Alhamdoosh, Gordon K. Smyth, and Matthew E. Ritchie. “<a href="https://doi.org/10.12688/f1000research.27893.1">A Guide to Creating Design Matrices for Gene Expression Experiments.</a>” F1000Research (December 10, 2020) - Design matrices for various experimental designs. Means model or mean-reference model.</p></li>
<li><p>Soneson, C, F Marini, F Geier, MI Love, and MB Stadler. “<a href="https://doi.org/10.12688/f1000research.24187.2">ExploreModelMatrix: Interactive Exploration for Improved Understanding of Design Matrices and Linear Models in R</a>” F1000Research, (June 4, 2020).</p></li>
</ul>
</div>
<div id="workflows-and-tools" class="section level3">
<h3>Workflows and tools</h3>
<ul>
<li><p><a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">Introduction to DGE</a> - differential expression analysis by DESeq2, by the teaching team at the Harvard Chan Bioinformatics Core (HBC)</p></li>
<li><p>He, Wen, Shanrong Zhao, Chi Zhang, Michael S. Vincent, and Baohong Zhang. “<a href="https://doi.org/10.1007/978-1-4939-7710-9_4">QuickRNASeq: Guide for Pipeline Implementation and for Interactive Results Visualization</a>.” Springer New York, 2018. - Practical RNA-seq tutorial based on QuickRNASeq publication.</p></li>
<li><p>Love, Michael I., Simon Anders, Vladislav Kim, and Wolfgang Huber. “<a href="https://doi.org/10.12688/f1000research.7035.2">RNA-Seq Workflow: Gene-Level Exploratory Analysis and Differential Expression</a>.” F1000Research (November 17, 2016) - RNA-seq workflow. From count import, including tximport, through EDA, DESeq2, batch removal, time course analysis, visualization.</p></li>
<li><p>Griffith, Malachi, Jason R. Walker, Nicholas C. Spies, Benjamin J. Ainscough, and Obi L. Griffith. “<a href="https://doi.org/10.1371/journal.pcbi.1004393">Informatics for RNA Sequencing: A Web Resource for Analysis on the Cloud</a>.” PLoS Computational Biology, (August 2015) - RNA-seq technology and analysis introduction. Very interesting are supplementary tables. <a href="https://rnabio.org/">The full tutorials</a></p></li>
<li><p>Sahraeian, Sayed Mohammad Ebrahim, Marghoob Mohiyuddin, Robert Sebra, Hagen Tilgner, Pegah T. Afshar, Kin Fai Au, Narges Bani Asadi, et al. “<a href="https://doi.org/10.1038/s41467-017-00050-4">Gaining Comprehensive Biological Insight into the Transcriptome by Performing a Broad-Spectrum RNA-Seq Analysis</a>.” Nature Communications, (December 2017) - RNAcocktail - RNA-seq tools benchmarking. All aspects of RNA-seq analysis, structured, Fig 1. Recommended tools Fig 8. <a href="https://bioinform.github.io/rnacocktail/">RNACocktail - A comprehensive framework for accurate and efficient RNA-Seq analysis</a>, <a href="http://www.rna-seqblog.com/unleash-the-power-within-rna-seq/">RNA-seq blog: Unleash the power within RNA-seq</a></p></li>
<li><p>Law, Charity W., Monther Alhamdoosh, Shian Su, Gordon K. Smyth, and Matthew E. Ritchie. “<a href="https://doi.org/10.12688/f1000research.9005.2">RNA-Seq Analysis Is Easy as 1-2-3 with Limma, Glimma and EdgeR</a>.” F1000Research (2016) - <a href="https://f1000research.com/articles/5-1408/v3">Latest Rsubread-limma plus pipeline</a>. <a href="https://bioconductor.org/packages/RNAseq123/">The complete R code for RNA-seq analysis tutorial</a></p></li>
<li><p>Pertea, Mihaela, Daehwan Kim, Geo M. Pertea, Jeffrey T. Leek, and Steven L. Salzberg. “<a href="https://doi.org/10.1038/nprot.2016.095">Transcript-Level Expression Analysis of RNA-Seq Experiments with HISAT, StringTie and Ballgown</a>.” Nature Protocols, (September 2016) - New Tuxedo suite. Protocol.</p></li>
<li><p>“<a href="https://youtu.be/96yBPM8lEt8">RNA-Seq Methods and Algorithms</a>” 7m video by Harold Pimentel, pseudoalignment, kallisto, sleuth, practical</p></li>
<li><p>Zhao, Qi, Yubin Xie, Peng Nie, Rucheng Diao, Licheng Sun, Zhixiang Zuo, and Jian Ren. “<a href="https://doi.org/10.1101/360461">IDEA: A Web Server for Interactive Differential Expression Analysis with R Packages</a>,” July 3, 2018. - Differential expression analysis from a matrix of FPKMs and a design matrix. Several methods to detect DEGs (DESeq2, edgeR, NOISeq, PoissonSeq, SAMseq), plots (MA, volcano, heatmap). <a href="http://renlab.org:3838/IDEA/">http://renlab.org:3838/IDEA/</a></p></li>
<li><p>TPMCalculator quantifies mRNA abundance directly from the alignments by parsing BAM files. The input parameters are the same GTF files used to generate the alignments, and one or multiple input BAM file(s) containing either single-end or paired-end sequencing reads. The TPMCalculator output is comprised of four files per sample reporting the TPM values and raw read counts for genes, transcripts, exons and introns respectively. <a href="https://github.com/ncbi/TPMCalculator">https://github.com/ncbi/TPMCalculator</a></p></li>
</ul>
</div>
<div id="rna-seq-resources" class="section level3">
<h3>RNA-seq resources</h3>
<ul>
<li><a href="http://journals.plos.org/ploscompbiol/article/file?type=supplementary&amp;id=info:doi/10.1371/journal.pcbi.1004393.s004">Tools for RNA-seq data analysis</a>
-<a href="https://github.com/crazyhottommy/RNA-seq-analysis">RNAseq analysis notes by Tommy Tang</a></li>
<li><a href="http://rnaseq.uoregon.edu/">University of Oregon’s RNA-seqlopedia</a>, a comprehensive guide to RNA-seq starting with experimental design, going through library prep, sequencing, and data analysis.</li>
<li>RNA-seq blog, <a href="http://www.rna-seqblog.com/">http://www.rna-seqblog.com/</a>, Several blog posts per week on new methods and tools for RNA-seq analysis</li>
<li><a href="https://github.com/griffithlab/rnaseq_tutorial">Informatics for RNA-seq, by Griffith lab</a></li>
</ul>
</div>
<div id="practicals" class="section level3">
<h3>Practicals</h3>
<ul>
<li><p><a href="https://bioconductor.org/packages/rnaseqGene/">RNA-seq workflow: gene-level exploratory analysis and differential expression</a> R package</p></li>
<li><p><a href="https://usegalaxy.org/u/jeremy/p/galaxy-rna-seq-analysis-exercise">RNA-seq analysis exercise using Galaxy</a>, an example analysis using the Tophat+Cufflinks workflow.</p></li>
<li><p><a href="https://github.com/chapmanb/bcbio-nextgen">“bcbio-nextgen” - Validated, scalable, community developed variant calling, RNA-seq and small RNA analysis</a></p></li>
<li><p><a href="https://davetang.org/muse/2017/10/25/getting-started-hisat-stringtie-ballgown/">Getting started with HISAT, StringTie, and Ballgown</a></p></li>
<li><p><a href="https://github.com/waldronlab/enrichOmics">“enrichOmics” - Functional enrichment analysis of high-throughput omics data</a>. From basic ExpressionSet differential and functional enrichment analysis to genomic region enrichment analysis and MultiAssayExperiment demo</p></li>
<li><p><a href="https://github.com/ENCODE-DCC/">ENCODE RNA-, ChIP-, DNAse-, ATAC- and other seq pipelines</a></p></li>
</ul>
</div>
