---
title: "Functional enrichment analysis"
# linktitle: "1. Introduction to RStudio and Images"
date: "2021-03-22"
# class_date: "2021-01-25"
output:
  blogdown::html_page:
    toc: false
menu:
  reading:
    parent: Reading
    weight: 11
type: docs
# weight: 1 
# pdf: 
# thumb: 
editor_options: 
  chunk_output_type: console
---

<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>


<ul>
<li><p>Reimand, Jüri, Ruth Isserlin, Veronique Voisin, Mike Kucera, Christian Tannus-Lopes, Asha Rostamianfar, Lina Wadi, et al. “<a href="https://doi.org/10.1038/s41596-018-0103-9">Pathway Enrichment Analysis and Visualization of Omics Data Using g:Profiler, GSEA, Cytoscape and EnrichmentMap</a>.” Nature Protocols, January 21, 2019 - Pathway enrichment analysis tutorial. Main tools - g:Profiler, GSEA, Cytoscape, EnrichmentMap. Definitions, step-by-step protocol.</p></li>
<li><p>Bard, Jonathan B. L., and Seung Y. Rhee. “<a href="https://doi.org/10.1038/nrg1295">Ontologies in Biology: Design, Applications and Future Challenges</a>.” Nature Reviews. Genetics, (March 2004) - Ontologies review</p></li>
<li><p>Ackermann, Marit, and Korbinian Strimmer. “<a href="https://doi.org/10.1186/1471-2105-10-47">A General Modular Framework for Gene Set Enrichment Analysis</a>.” BMC Bioinformatics, (2009) - All steps for enrichment analysis, methods, statistics, GSEA.</p></li>
<li><p>Efron, Bradley, and Robert Tibshirani. “<a href="https://doi.org/10.1214/07-AOAS101">On Testing the Significance of Sets of Genes</a>.” The Annals of Applied Statistics, 2007 - maxmean statistics for enrichment analysis. Comparison with GSEA.</p></li>
<li><p>Huang, Da Wei, Brad T. Sherman, and Richard A. Lempicki. “<a href="https://doi.org/10.1093/nar/gkn923">Bioinformatics Enrichment Tools: Paths toward the Comprehensive Functional Analysis of Large Gene Lists</a>.” Nucleic Acids Research, (January 2009) - Gene enrichment analyses tools. Statistics, concept of background. 68 tools, table</p></li>
<li><p>Hung, Jui-Hung, Tun-Hsiang Yang, Zhenjun Hu, Zhiping Weng, and Charles DeLisi. “<a href="https://doi.org/10.1093/bib/bbr049">Gene Set Enrichment Analysis: Performance Evaluation and Usage Guidelines</a>.” Briefings in Bioinformatics, (May 2012) - Details of GSEA. Statistics, correction for multiple testing. Lack of gold standard - concept of mutual coverage.</p></li>
<li><p>Khatri, Purvesh, Marina Sirota, and Atul J. Butte. “<a href="https://doi.org/10.1371/journal.pcbi.1002375">Ten Years of Pathway Analysis: Current Approaches and Outstanding Challenges</a>.” PLoS Computational Biology, (2012) - Review of enrichment analyses techniques, focus on pathways, Table 1 lists tools, limitations.</p></li>
<li><p>Hänzelmann, Sonja, Robert Castelo, and Justin Guinney. “<a href="https://doi.org/10.1186/1471-2105-14-7">GSVA: Gene Set Variation Analysis for Microarray and RNA-Seq Data</a>.” BMC Bioinformatics (January 16, 2013) - GSVA - a GSE method that estimates variation of pathway activity over a sample population in an unsupervised manner</p></li>
<li><p>Khatri, Purvesh, Marina Sirota, and Atul J. Butte. “<a href="https://doi.org/10.1371/journal.pcbi.1002375">Ten Years of Pathway Analysis: Current Approaches and Outstanding Challenges</a>.” PLoS Computational Biology, (February 23, 2012) - Overview of pathway analysis methods (ORA - over-representation analysis, FCS - functional class scoring, PT - pathway topology) and tools. Methods, limitations.</p></li>
<li><p>Nam D., and Seon-Young K.. “<a href="https://doi.org/10.1093/bib/bbn001">Gene-Set Approach for Expression Pattern Analysis</a>.” Briefings in Bioinformatics 2008</p></li>
<li><p>Mutation Consequences and Pathway Analysis working group. “<a href="https://doi.org/10.1038/nmeth.3440">Pathway and Network Analysis of Cancer Genomes</a>.” Nature Methods 2015</p></li>
<li><p>de Leeuw, C. et.al. “<a href="https://doi.org/10.1038/nrg.2016.29">The Statistical Properties of Gene-Set Analysis</a>.” Nature Reviews 2016</p></li>
</ul>
