---
title: "Epigenomics, methylation"
# linktitle: "1. Introduction to RStudio and Images"
date: "2021-04-07"
# class_date: "2021-01-25"
output:
  blogdown::html_page:
    toc: false
menu:
  reading:
    parent: Reading
    weight: 13
type: docs
# weight: 1 
# pdf: 
# thumb: 
editor_options: 
  chunk_output_type: console
---

## Epigenomics intro

- Stephen B. Baylin and Peter A. Jones, “[A Decade of Exploring the Cancer Epigenome — Biological and Translational Implications](https://doi.org/10.1038/nrc3130),” Nature Reviews Cancer, (September 23, 2011) - Cancer epigenomics introduction, therapies

- Kagohara, Luciane T., Genevieve L. Stein-O’Brien, Dylan Kelley, Emily Flam, Heather C. Wick, Ludmila V. Danilova, Hariharan Easwaran, et al. “[Epigenetic Regulation of Gene Expression in Cancer: Techniques, Resources and Analysis](https://doi.org/10.1093/bfgp/elx018).” Briefings in Functional Genomics, August 11, 2017 - Review of epigeneitc modifications, methylation, histones, chromatin states, 3D. Technologies, databases, software. Lots of references

- Li, Bing, Michael Carey, and Jerry L. Workman. “[The Role of Chromatin during Transcription](http://www.cell.com/abstract/S0092-8674(07)00109-2).” Cell, (February 23, 2007) - Transcription process and the role of chromatin modifications. 

- Zhou, Vicky W., Alon Goren, and Bradley E. Bernstein. “[Charting Histone Modifications and the Functional Organization of Mammalian Genomes](https://www.nature.com/articles/nrg2905).” Nature Reviews. Genetics, (January 2011) - Histone marks review, ChIP-seq. Graphics of histone marks roles

- Wang, Zhibin, Dustin E. Schones, and Keji Zhao. “[Characterization of Human Epigenomes](https://doi.org/10.1016/j.gde.2009.02.001).” Current Opinion in Genetics & Development, (April 2009) - Concise description of main histone marks, their roles in transcription, and the corresponding studies. Figure 2 - schematic distribution of histone marks with respect to genes-TSSs.

- Zhang, Z. D., A. Paccanaro, Y. Fu, S. Weissman, Z. Weng, J. Chang, M. Snyder, and M. B. Gerstein. “[Statistical Analysis of the Genomic Distribution and Correlation of Regulatory Elements in the ENCODE Regions](https://doi.org/10.1101/gr.5573107).” Genome Research, (June 1, 2007) - ENCODE pilot project analysis. Non-random location of regulatory elements. Enrichment in TSSs, not in the middle or end of transcription sites. PCA and biplot representation of interrelatedness among TFs and histone marks, clustering.

- Huen, David S., and Steven Russell. “[On the Use of Resampling Tests for Evaluating Statistical Significance of Binding-Site Co-Occurrence](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-359).” BMC Bioinformatics (June 30, 2010) - Thorough review on permutation. Problem with different numbers of ROIs/epiregions in permutations. Regions may overlap (i.e., occur in clusters). Independent assignment is best.

## Epigenomic enrichment

- McLean, Cory Y., Dave Bristor, Michael Hiller, Shoa L. Clarke, Bruce T. Schaar, Craig B. Lowe, Aaron M. Wenger, and Gill Bejerano. “[GREAT Improves Functional Interpretation of Cis-Regulatory Regions](https://doi.org/10.1038/nbt.1630).” Nature Biotechnology 28, no. 5 (May 2010) - Hypergeometric and binomial enrichment of regulatory regions in relation to genesgenomic regions and their ontologies.
