---
title: "Epigenomics, methylation"
# linktitle: "1. Introduction to RStudio and Images"
date: "2021-04-07"
# class_date: "2021-01-25"
output:
  blogdown::html_page:
    toc: false
menu:
  reading:
    parent: Reading
    weight: 13
type: docs
# weight: 1 
# pdf: 
# thumb: 
editor_options: 
  chunk_output_type: console
---



<div id="epigenomics-intro" class="section level2">
<h2>Epigenomics intro</h2>
<ul>
<li><p>Stephen B. Baylin and Peter A. Jones, “<a href="https://doi.org/10.1038/nrc3130">A Decade of Exploring the Cancer Epigenome — Biological and Translational Implications</a>,” Nature Reviews Cancer, (September 23, 2011) - Cancer epigenomics introduction, therapies</p></li>
<li><p>Kagohara, Luciane T., Genevieve L. Stein-O’Brien, Dylan Kelley, Emily Flam, Heather C. Wick, Ludmila V. Danilova, Hariharan Easwaran, et al. “<a href="https://doi.org/10.1093/bfgp/elx018">Epigenetic Regulation of Gene Expression in Cancer: Techniques, Resources and Analysis</a>.” Briefings in Functional Genomics, August 11, 2017 - Review of epigeneitc modifications, methylation, histones, chromatin states, 3D. Technologies, databases, software. Lots of references</p></li>
<li><p>Li, Bing, Michael Carey, and Jerry L. Workman. “<a href="http://www.cell.com/abstract/S0092-8674(07)00109-2">The Role of Chromatin during Transcription</a>.” Cell, (February 23, 2007) - Transcription process and the role of chromatin modifications.</p></li>
<li><p>Zhou, Vicky W., Alon Goren, and Bradley E. Bernstein. “<a href="https://www.nature.com/articles/nrg2905">Charting Histone Modifications and the Functional Organization of Mammalian Genomes</a>.” Nature Reviews. Genetics, (January 2011) - Histone marks review, ChIP-seq. Graphics of histone marks roles</p></li>
<li><p>Wang, Zhibin, Dustin E. Schones, and Keji Zhao. “<a href="https://doi.org/10.1016/j.gde.2009.02.001">Characterization of Human Epigenomes</a>.” Current Opinion in Genetics &amp; Development, (April 2009) - Concise description of main histone marks, their roles in transcription, and the corresponding studies. Figure 2 - schematic distribution of histone marks with respect to genes-TSSs.</p></li>
<li><p>Zhang, Z. D., A. Paccanaro, Y. Fu, S. Weissman, Z. Weng, J. Chang, M. Snyder, and M. B. Gerstein. “<a href="https://doi.org/10.1101/gr.5573107">Statistical Analysis of the Genomic Distribution and Correlation of Regulatory Elements in the ENCODE Regions</a>.” Genome Research, (June 1, 2007) - ENCODE pilot project analysis. Non-random location of regulatory elements. Enrichment in TSSs, not in the middle or end of transcription sites. PCA and biplot representation of interrelatedness among TFs and histone marks, clustering.</p></li>
<li><p>Huen, David S., and Steven Russell. “<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-359">On the Use of Resampling Tests for Evaluating Statistical Significance of Binding-Site Co-Occurrence</a>.” BMC Bioinformatics (June 30, 2010) - Thorough review on permutation. Problem with different numbers of ROIs/epiregions in permutations. Regions may overlap (i.e., occur in clusters). Independent assignment is best.</p></li>
</ul>
</div>
<div id="epigenomic-enrichment" class="section level2">
<h2>Epigenomic enrichment</h2>
<ul>
<li><p>McLean, Cory Y., Dave Bristor, Michael Hiller, Shoa L. Clarke, Bruce T. Schaar, Craig B. Lowe, Aaron M. Wenger, and Gill Bejerano. “<a href="https://doi.org/10.1038/nbt.1630">GREAT Improves Functional Interpretation of Cis-Regulatory Regions</a>.” Nature Biotechnology 28, no. 5 (May 2010) - Hypergeometric and binomial enrichment of regulatory regions in relation to genesgenomic regions and their ontologies.</p></li>
<li><p>De, Subhajyoti, Brent S. Pedersen, and Katerina Kechris. “<a href="https://doi.org/10.1093/bib/bbt053">The Dilemma of Choosing the Ideal Permutation Strategy While Estimating Statistical Significance of Genome-Wide Enrichment</a>.” Briefings in Bioinformatics 15, no. 6 (November 2014)</p></li>
<li><p>Dozmorov, Mikhail G. “<a href="https://doi.org/10.1093/bioinformatics/btx414">Epigenomic Annotation-Based Interpretation of Genomic Data: From Enrichment Analysis to Machine Learning</a>.” Bioinformatics (Oxford, England) 33, no. 20 (October 15, 2017)</p></li>
</ul>
<div id="hidden-markov-models-genome-segmentation-overview" class="section level3">
<h3>Hidden Markov Models, genome segmentation overview</h3>
<ul>
<li><p>Eddy, Sean R. “<a href="https://doi.org/10.1038/nbt1004-1315">What Is a Hidden Markov Model?</a>” Nature Biotechnology, (October 2004)</p></li>
<li><p>Eddy, S. R. “<a href="https://pdfs.semanticscholar.org/bc1e/ffe17026c1396697e4ea2b399f7a049202fc.pdf">Multiple Alignment Using Hidden Markov Models</a>.” Proceedings. International Conference on Intelligent Systems for Molecular Biology (1995) - HMMER hidden markov models for protein sequence alignment. First paper</p></li>
<li><p>Schuster-Böckler, Benjamin, and Alex Bateman. “<a href="https://doi.org/10.1002/0471250953.bia03as18">An Introduction to Hidden Markov Models</a>.” 2007 - Hidden Markov Models primer</p></li>
<li><p>L.R. Rabiner, “<a href="https://doi.org/10.1109/5.18626">A Tutorial on Hidden Markov Models and Selected Applications in Speech Recognition</a>,” Proceedings of the IEEE, (February 1989) - Theory and statistics of Hidden Markov Models. Very detailed, thorough</p></li>
<li><p>Ernst, Jason, and Manolis Kellis. “<a href="https://doi.org/10.1038/nprot.2017.124">Chromatin-State Discovery and Genome Annotation with ChromHMM</a>.” Nature Protocols, (December 2017) - ChromHMM protocol. Intro about ChromHMM, other methods. Links to genome annotation models.</p></li>
<li><p>Hoffman, Michael M., Orion J. Buske, Jie Wang, Zhiping Weng, Jeff A. Bilmes, and William Stafford Noble. “<a href="https://doi.org/10.1038/nmeth.1937">Unsupervised Pattern Discovery in Human Chromatin Structure through Genomic Segmentation</a>.” Nature Methods 9, no. 5 (May 2012) - Segway - segmentation and prediction of genomic states. Using dynamic Bayesian network</p></li>
</ul>
</div>
<div id="methylation" class="section level3">
<h3>Methylation</h3>
<ul>
<li><p>Pidsley, R., et. al., and Susan J. Clark. “<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1066-1">Critical Evaluation of the Illumina MethylationEPIC BeadChip Microarray for Whole-Genome DNA Methylation Profiling</a>.” Genome Biology 2016</p></li>
<li><p>Pan D., et. al. “<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-587">Comparison of Beta-value and M-value methods for quantifying methylation levels by microarray analysis</a>.” BMC Bioinformatics, 2010</p></li>
<li><p>Bock, Christoph, Eleni M Tomazou, Arie B Brinkman, Fabian Müller, Femke Simmer, Hongcang Gu, Natalie Jäger, Andreas Gnirke, Hendrik G Stunnenberg, and Alexander Meissner. “<a href="https://doi.org/10.1038/nbt.1681">Quantitative Comparison of Genome-Wide DNA Methylation Mapping Technologies</a>.” Nature Biotechnology, (October 2010) - Methylation intro, technology. Software tools, tables. Quality control and problems. Differential analysis. Public repositories.</p></li>
<li><p>Krueger, Felix, Benjamin Kreck, Andre Franke, and Simon R Andrews. “<a href="https://www.nature.com/articles/nmeth.1828">DNA Methylome Analysis Using Short Bisulfite Sequencing Data</a>.” Nature Methods, (January 30, 2012) - Methylation intro, technologies to measure. Alignment problems, QC considerations, processing workflow. Theoretical, references.</p></li>
<li><p>Wreczycka, Katarzyna, Alexander Gosdschan, Dilmurat Yusuf, Bjoern Gruening, Yassen Assenov, and Altuna Akalin. “<a href="https://www.sciencedirect.com/science/article/pii/S0168165617315936">Strategies for Analyzing Bisulfite Sequencing Data</a>,” August 9, 2017 - Bisufite sequencing data analysis steps. Intro into methylation. Refs to packages.</p></li>
<li><p>Robinson, Mark D., Abdullah Kahraman, Charity W. Law, Helen Lindsay, Malgorzata Nowicka, Lukas M. Weber, and Xiaobei Zhou. “<a href="https://doi.org/10.3389/fgene.2014.00324">Statistical Methods for Detecting Differentially Methylated Loci and Regions</a>.” Frontiers in Genetics (2014) - Methylation review, technology, databases, experimental design, statistics and tools for differential methylation detection, beta-binomial distribution, cell type deconvolution.</p></li>
<li><p>Krueger, Felix, and Simon R. Andrews. “<a href="https://doi.org/10.1093/bioinformatics/btr167">Bismark: A Flexible Aligner and Methylation Caller for Bisulfite-Seq Applications</a>.” Bioinformatics, (June 1, 2011) - Bismark paper. stranded and unstranded BS sequencing. Conversion of reads, genomes, best alignment strategy. <a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/bismark/</a></p></li>
<li><p>Xi, Yuanxin, and Wei Li. “<a href="https://doi.org/10.1186/1471-2105-10-232">BSMAP: Whole Genome Bisulfite Sequence MAPping Program</a>.” BMC Bioinformatics, (2009) - BSMAP paper. Bisulphite conversion technology introduction, problems. BSMAP algorithm. Very good figures explaining all steps. <a href="https://code.google.com/archive/p/bsmap/">BSMAP software</a></p></li>
<li><p>Chen, Yunshun, Bhupinder Pal, Jane E. Visvader, and Gordon K. Smyth. “<a href="https://doi.org/10.12688/f1000research.13196.1">Differential Methylation Analysis of Reduced Representation Bisulfite Sequencing Experiments Using EdgeR</a>.” F1000Research (November 28, 2017) - RRBS differential methylation analysis. Methylation intro. R code tutorial.</p></li>
<li><p>Teschendorff, Andrew E., and Caroline L. Relton. “<a href="https://doi.org/10.1038/nrg.2017.86">Statistical and Integrative System-Level Analysis of DNA Methylation Data</a>.” Nature Reviews Genetics, November 13, 2017 - Deconvolution of methylation profiles. Reference-based, reference-free, semi-reference-free. Table 1 - tools</p></li>
<li><p>Methylation statistics packages: Table 2 in Liu, Hongbo, Song Li, Xinyu Wang, Jiang Zhu, Yanjun Wei, Yihan Wang, Yanhua Wen, et al. “<a href="https://www.ncbi.nlm.nih.gov/pubmed/27515490">DNA Methylation Dynamics: Identification and Functional Annotation</a>.” Briefings in Functional Genomics, 2016</p></li>
<li><p>Mark D. Robinson et al., “<a href="https://doi.org/10.3389/fgene.2014.00324">Statistical Methods for Detecting Differentially Methylated Loci and Regions</a>,” Frontiers in Genetics 5 (2014) - Methylation nethods review. From data, experimental design to software tools for finding differentially methylated regions.</p></li>
<li><p>Shafi, Adib, Cristina Mitrea, Tin Nguyen, and Sorin Draghici. “<a href="https://doi.org/10.1093/bib/bbx013">A Survey of the Approaches for Identifying Differential Methylation Using Bisulfite Sequencing Data</a>.” Briefings in Bioinformatics, March 8, 2017 - Review of differential methylation methods and 22 tools. Categorized by approaches. <a href="https://academic.oup.com/view-large/figure/121909042/bbx013f3.tif">Pros and cons of each approach</a>, <a href="https://academic.oup.com/view-large/121909044">Table 1. Summary of the important characteristics of the 22 surveyed approaches</a>, <a href="https://academic.oup.com/view-large/121909057">Table 2. Comparison of the available implementations of the 22 surveyed approaches</a></p></li>
<li><p>Wreczycka, Katarzyna, Alexander Gosdschan, Dilmurat Yusuf, Björn Grüning, Yassen Assenov, and Altuna Akalin. “<a href="https://doi.org/10.1016/j.jbiotec.2017.08.007">Strategies for Analyzing Bisulfite Sequencing Data</a>.” Journal of Biotechnology (November 2017) - Review of bisulfite sequencing technology, data preprocessing, and analysis methods and tools. Differential methylation.</p></li>
<li><p>Venet, D., F. Pecasse, C. Maenhaut, and H. Bersini. “<a href="https://academic.oup.com/bioinformatics/article/17/suppl_1/S279/262438">Separation of Samples into Their Constituents Using Gene Expression Data</a>.” Bioinformatics (2001) - Statistical derivation of deconvolution.</p></li>
</ul>
</div>
</div>
