---
title: "Genomic variation"
# linktitle: "1. Introduction to RStudio and Images"
date: "2021-04-28"
# class_date: "2021-01-25"
output:
  blogdown::html_page:
    toc: false
menu:
  reading:
    parent: Reading
    weight: 17
type: docs
# weight: 1 
# pdf: 
# thumb: 
editor_options: 
  chunk_output_type: console
---

<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>


<div id="single-nucleotide-polymorphisms" class="section level2">
<h2>Single Nucleotide Polymorphisms</h2>
<ul>
<li><p>Koboldt, Daniel C. “<a href="https://doi.org/10.1186/s13073-020-00791-w">Best Practices for Variant Calling in Clinical Sequencing</a>.” Genome Medicine, (December 2020) - Introduction in genomic variant calling, panel/exome/whole genome sequencing technologies (Table 1), preprocessing, analysis (SNVs/indels, mutations, CNVs, SVs, gene fusions, Table 2), gold standard datasets (GIAB), best practices, filtering for each type of genomic variant.</p></li>
<li><p>1000 Genomes Project Consortium, Adam Auton, Lisa D. Brooks, Richard M. Durbin, Erik P. Garrison, Hyun Min Kang, Jan O. Korbel, et al. “<a href="https://www.nature.com/nature/journal/v526/n7571/full/nature15393.html">A Global Reference for Human Genetic Variation</a>.” Nature 526, no. 7571 (October 1, 2015)</p></li>
<li><p>Bamshad, Michael J., Sarah B. Ng, Abigail W. Bigham, Holly K. Tabor, Mary J. Emond, Deborah A. Nickerson, and Jay Shendure. “<a href="https://www.nature.com/articles/nrg3031">Exome Sequencing as a Tool for Mendelian Disease Gene Discovery</a>.” Nature Reviews. Genetics 12, no. 11 (September 27, 2011) - Exome sequencing technology, limitations, use for diagnostics, family studies.</p></li>
<li><p>McCarthy, Davis J, Peter Humburg, Alexander Kanapin, Manuel A Rivas, Kyle Gaulton, asds, Jean-Baptiste Cazier, and Peter Donnelly. “<a href="https://genomemedicine.biomedcentral.com/articles/10.1186/gm543">Choice of Transcripts and Software Has a Large Effect on Variant Annotation</a>.” Genome Medicine, (2014) - SNP annotation depends on transcripts and software. Types of SNPs. Ambigious annotations</p></li>
<li><p>Pabinger, S., A. Dander, M. Fischer, R. Snajder, M. Sperk, M. Efremova, B. Krabichler, M. R. Speicher, J. Zschocke, and Z. Trajanoski. “<a href="https://doi.org/10.1093/bib/bbs086">A Survey of Tools for Variant Analysis of Next-Generation Genome Sequencing Data</a>.” Briefings in Bioinformatics, (March 1, 2014) - SNP calling and analysis tools overview. Germline, somatic, CNV, SV detection. Variant annotation tools.</p></li>
<li><p>MacArthur, D. G., T. A. Manolio, D. P. Dimmock, H. L. Rehm, J. Shendure, G. R. Abecasis, D. R. Adams, et al. “<a href="https://doi.org/10.1038/nature13127">Guidelines for Investigating Causality of Sequence Variants in Human Disease</a>.” Nature, (April 24, 2014) - Definitions and guidelines to define pathogenicity of SNPs</p></li>
</ul>
</div>
