---
title: "RNA-seq Introduction"
author: "Mikhail Dozmorov"
date: "2021-03-03"
output:
  beamer_presentation:
    colortheme: dolphin
    fig_caption: no
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    theme: AnnArbor
  ioslides_presentation:
    css:
    - css/fonts.css
    - css/custom.css
    - css/title-slide.css
    - css/slide-background.css
    transition: faster
    widescreen: true
    includes:
      before_body: html/title.html
---

# Introduction

## Eukaryotic gene structure

\includegraphics[height=160px]{img/gene_structure.png}

## Gene expression

\includegraphics[height=200px]{img/central_dogma1.png}

## Gene expression

\includegraphics[height=200px]{img/central_dogma2.png}

## What is RNA sequencing?

- Massive parallel sequencing to **characterize and quantify transcriptomes** (all actively transcribed genes)
- Detection of **differential gene expression**
- **Transcriptome reconstruction**, identification of **new transcripts**
- Detection of **alternative splicing events**
- Detection of **structural variants**, e.g., fusion transcripts
- **Allele-specific** gene expression measurements
- **Mutation analysis** – presence of genomic mutations and their effect on gene expression

\tiny [RNA-seq analysis techniques](http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s003)

## Sequencing technologies

Commercially available

- **Illumina/Solexa** - short reads, sequencing-by-synthesis
- **Life Technologies Ion Torrent/Proton** - short reads, Ion Semiconductor sequencing
- **Pacific Biosciences** - long reads, Single Molecule Real Time sequencing

Experimental

- **Nanopore sequencing** - continuous sequencing (very long reads), fluctuations of the ionic current from nucleotides passing through the nanopore

<!--
## Cap Analysis of Gene Expression (CAGE)

- The first 20 to 21 nucleotides at the 5' ends of capped mRNAs are extracted by a combination of cap trapping and cleavage by restriction enzyme _MmeI_.
- The "CAGE tags" then sequenced and mapped to the genome
- Produces a genome-wide picture of transcription start sites at a base-pair resolution
- "Promoterome" generated by FANTOM project

\tiny Balwierz, Piotr J., Piero Carninci, Carsten O. Daub, Jun Kawai, Yoshihide Hayashizaki, Werner Van Belle, Christian Beisel, and Erik van Nimwegen. “Methods for Analyzing Deep Sequencing Expression Data: Constructing the Human and Mouse Promoterome with DeepCAGE Data.” Genome Biology 10, no. 7 (2009): R79. doi:10.1186/gb-2009-10-7-r79. https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-7-r79
-->
## Overview of RNA sequencing technology

\includegraphics[height=200px]{img/rna-seq_tech.png}

\tiny Source: http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004393

<!--
## Advantage of RNA-Seq over Microarray

- Much richer information beyond quantitation
    - Boundary of gene transcripts: both 5' and 3' end, to nucleotide level 
    - Alternative exon usage, novel splicing junction detection
    - SNP/indel discovery in transcripts: both coding and UTRs
    - Allele specific expression: critical in imprinting, cancer

&nbsp;

- Not relying on gene annotation by mapping to the whole genome
    - No longer biased by probe design
    - Novel gene and exon discovery enabled

## Advantage of RNA-Seq over Microarray

- Better performance at quantitation
    - Unlimited dynamic range: by increasing depth as needed
    - Higher specificity and accuracy: digital counts of transcript copies, very low background noise
    - Higher sensitivity: more transcripts and more differential genes detected

&nbsp;

- Re-analysis easily done by computation, as gene annotation keeps evolving 
- _De novo_ assembly possible, not relying on reference genome sequence 
- Comparable cost, continuing to drop


## SEQC RNA-seq quality control.

- The third phase of the MicroArray Quality Control (MAQC) project, also known as the Sequencing Quality Control (SEQC) project, aims to assess the technical performance of high-throughput sequencing platforms by generating benchmarking data sets
- The design includes four different sample types, namely samples A, B, C and D
    - Sample A is Stratagene’s universal human reference (UHR) RNA
    - Sample B is Ambion’s human brain reference RNA
    - Samples C and D are mixes of samples A and B, in a 3:1 and 1:3 ratio, respectively
- The four reference samples were sent to several sequencing centers around the world and sequenced using different platforms
- Found good reproducibility across sites and platforms

\tiny SEQC/MAQC-III Consortium. A comprehensive assessment of RNA-seq accuracy, reproducibility and information content by the Sequencing Quality Control Consortium. Nat. Biotechnol. doi:10.1038/nbt.2957 (24 August 2014).
-->

## RNA-Seq Limitations

Quantitation influenced by many confounding factors  

- "Sequenceability" - varying across genomic regions, local GC content and structure-related
- Varying length of gene transcripts and exons
- Bias in read ends due to reverse transcription, subtle but consistent 
- Varying extent of PCR amplification artifacts
- Effect of RNA degradation in the real world
- Computational bias in aligning reads to genome due to aligners
    
## RNA-Seq Limitations

SNP discovery in RNA-seq is more challenging than in DNA  

- Varying levels of coverage depth
- False discovery around splicing junctions due to incorrect mapping
    
_De novo_ assembly of transcripts without genome sequence: computationally intensive but possible, technical improvements will help  

- Longer read length
- Lower error rate
- More uniform nucleotide coverage of transcripts - more equalized transcript abundance

# Library preparation

## Library preparation steps

- **RNA isolation and QC**, to extract RNA relevant to the experimental question
- **Fragmentation**, to recover short reads across full length of long genes
- **Size selection**, suitable for RNA sequencing. 300-500bp - mRNA, 20-150bp - small/miRNA
- **Amplification**, typically by PCR. Up to $0.5-10ng$ of RNA
- **Library normalization/Exome capture**
- **Barcoding** and **multiplexing**
- Optionally, add **External RNA Control Consortium (ERCC) spike-in controls**
- **Single** or **paired end** sequencing. The latter is preferrable for the _de novo_ transcript discovery or isoform expression analysis

\tiny Sample prepatation and library construction strategies: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s005

## RNA isolation

- **Ribosomal RNA (rRNA) depletion**
    - $0.1-1\mu g$ original total RNA (One cell contains ~10 picogram of total RNA)
    - rRNAs constitute over 90 % of total RNA in the cell, leaving the 1–2 % comprising messenger RNA (mRNA) that we are normally interested in (One cell contains ~0.1 picogram mRNA)
    - Enriches for mRNA + long noncoding RNA. 
    - Hybridization to bead-bound rRNA probes

## RNA isolation

- **Poly(A) selection (for eukaryotes only)**
    - Enrich for mRNA.
    - Hybridization to oligo-dT beads

- **Small RNA extraction**
    - Specific kits required to retain small RNAs
    - Optionally, size-selection by gel

&nbsp;

\tiny Description of RNA-seq library enrichment strategies: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s006

## Poly-A selection or ribosome depletion protocol?

- Poly-A excels at gene quantification for classification/prediction purposes, better represents total RNA content
- Ribosomal depletion - more noncoding RNAs, better alignment of reads, more gene fusion events
- Overall, comparable performance

&nbsp;

\tiny Detailed comparison of RNA-seq library construction protocols: https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4039-1

## RNA quality

Agilent 2100 bioanalyzer. RIN - RNA integrity number (should be >7)

\includegraphics[height=200px]{img/agilent.png}

## Unstranded vs. Strand-specific library

- **Unstranded**:  Random hexamer priming to reverse-transcribe mRNA
- **Stranded**:  dUTP method - incorporating UTP nucleotides during the second cDNA synthesis, followed by digestion of the strand containing dUTP

&nbsp;

\tiny Strand-related settings for RNA-seq tools: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s007}

## Unstranded vs. Strand-specific library

\includegraphics[height=200px]{img/strand.png}


<!--
## External RNA Control Consortium (ERCC) spike-in controls

- A set of RNA standards for RNA-seq
    - 92 polyadenylated transcripts that mimic natural eukaryotic mRNAs
    - Designed to have a wide range of lengths (250–2,000 nucleotides) and GC-contents (5–51%) and can be spiked into RNA samples before library preparation at various concentrations (106-fold range)

- ERCC spike-in controls can be used for normalization in the context of a global expression shift
    - Count the number of cells in each sample
    - Add the ERCC spike-in sequences to each sample in proportion to the number of cells
    - Normalize read counts based on cyclic loess robust local regression on the spike-in counts

\tiny Baker, S.C. et al. The external RNA controls consortium: a progress report. Nat. Methods 2, 731–734 (2005).

Jiang, L. et al. Synthetic spike-in standards for RNA-seq experiments. Genome Res. 21, 1543–1551 (2011).

Loven, J. et al. Revisiting global gene expression analysis. Cell 151, 476–482
(2012).
-->

