---
title: "SummarizedExperiment"
author: "[Michael Love](http://mikelove.github.io)"
output: html_document
---

When I previously covered [objects](objects.html), we learned about
ExpressionSet, and mentioned that a similarly shaped object is
SummarizedExperiment. By shape, I refer to the 
[3 tables](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4509590/figure/F2/)
that make up the object, one for the data, one for the sample
information, and one for the feature/row information. In
SummarizedExperiment, these are called `assay`, `colData` and
`rowData` or `rowRanges` (more on this below). 
This generalizes the language that was used for
ExpressionSet, so they apply to more types of datasets (not just gene
expression: `exprs`, `phenoData`, and `featureData`).

It helps to start by making a small toy SummarizedExperiment, to see
how the pieces come together.

```{r echo=FALSE}
knitr::opts_chunk$set(cache=TRUE)
```

```{r message=FALSE}
library(SummarizedExperiment)
colData <- data.frame(sample=factor(1:6),
                      condition=factor(c("A","A","B","B","C","C")),
                      treated=factor(rep(0:1,3)))
colData
```

So far, it is similar to ExpressionSet. The key difference is that the
rows of a SummarizedExperiment can correspond to
[GRanges](ranges.html). If we don't have ranges, we can just put a
table on the side of the SummarizedExperiment by specifying `rowData`.

I will show in the example though how to provide `rowRanges`. Let's
use the first 10 genes in the Ensembl database for human:

```{r message=FALSE}
library(EnsDb.Hsapiens.v86)
txdb <- EnsDb.Hsapiens.v86
g <- genes(txdb)
g <- keepStandardChromosomes(g, pruning.mode="coarse")
rowRanges <- g[1:10]
```

We will make up some simulated "expression" measurements, and then
store these in the SummarizedExperiment. I call `list` so I can name
the matrix, otherwise it would not be named.

```{r}
exprs <- matrix(rnorm(6 * 10), ncol=6, nrow=10)
se <- SummarizedExperiment(assay=list("exprs"=exprs),
                           colData=colData,
                           rowRanges=rowRanges)
se
```

We see this object has one named matrix. It could have others, and in
that case you could access the first with `assay` and in general by
name, e.g. `assays(se)[["exprs"]]`

```{r}
assayNames(se)
```

Finally, if we wanted to add data onto the rows, for example, the
score of a test on the matrix data, we use the metadata columns
function, or `mcols`:

```{r}
mcols(se)$score <- rnorm(10)
mcols(se)
```

# Using the ranges of a SummarizedExperiment

How does this additional functionality of the `rowRanges` facilitate
faster data analysis? Suppose we are working with another data set
besides `se` and we find a region of interest on chromsome 1. If we
want to pull out the expression data for that region, we just ask for
the subset of `se` that overlaps. First we build the query region, and
then use the *GRanges* function `overlapsAny` within single square
brackets (like you would subset any matrix-like object:

```{r}
query <- GRanges("1", IRanges(25000,40000))
se.sub <- se[overlapsAny(se, query), ]
```

We get just three ranges, and three rows of the SummarizedExperiment:

```{r}
rowRanges(se.sub)
assay(se.sub)
```

Another useful property is that we know metadata about the
chromosomes, and the version of the genome. (If you were not yet
aware, the basepair position of a given feature, say gene *XYZ*, will
change between versions of the genome, as sequences are added or
rearranged --- you should also know that the human genome has not been
fully sequenced as of 2017, which we will discuss in class.)

```{r}
seqinfo(se)
```

# Downloading SummarizedExperiment data

We previously introduced the computational project, called 
[recount2](https://jhubiostatistics.shinyapps.io/recount/), which performs a
basic summarization of public data sets with gene expression data. We
will use data from *recount2* again.

This dataset contains RNA-seq samples from human airway epithelial
cell cultures. The paper is 
[here](https://www.ncbi.nlm.nih.gov/pubmed/25706956).
The structure of the experiment was that, cell cultures from 6
asthmatic and 6 non-asthmatics donors were treated with viral
infection or left untreated (controls). So we have 2 samples (control
or treated) for each of the 12 donors.

```{r}
url <- "http://duffel.rail.bio/recount/SRP046226/rse_gene.Rdata"
file <- "asthma.rda"
if (!file.exists(file)) download.file(url, file)
load(file)
```

We use a custom function to produce a matrix which a count of RNA
fragments for each gene (rows) and each sample (columns)

```{r}
library(here)
source(here("bioc","my_scale_counts.R"))
rse <- my_scale_counts(rse_gene)
```

We can take a peek at the column data:

```{r}
colData(rse)[,1:6]
```

The information we are interested in is contained in the
`characteristics` column (which is a character list).

```{r}
class(rse$characteristics)
rse$characteristics[1:3]
rse$characteristics[[1]]
```

We can pull out the 3 and 4 element using the `sapply` function and
the square bracket function. I know this syntax looks a little funny,
but it's really just saying, use the single square bracket, pull out
the third element (or fourth element). 

```{r}
rse$condition <- sapply(rse$characteristics, `[`, 3)
rse$treatment <- sapply(rse$characteristics, `[`, 4)
```

Let's see what the `rowRanges` of this experiment look like:

```{r}
rowRanges(rse)
seqinfo(rse)
```

The `rowRanges` here were determined by the quantification method 
that the *recount2* authors used. We don't know what the genome is 
from the `seqinfo`, but we could look this up from the project website.

The following code I use to clean up the condition and treatment variables:

```{r}
library(magrittr)
rse$condition
rse$condition %<>% (function(x) {
  factor(sub("-",".", sub("disease state: (.*)","\\1",x) ))
  })
rse$treatment %<>% (function(x) factor(sub("treatment: (.*)","\\1",x)))
```

Now we have:

```{r}
rse$condition
rse$treatment
```

# Visualizing count matrix data in a SummarizedExperiment

We will discuss transformations and normalization in a following 
section, but here we will just use a transformation so that we can 
compute meaningful distances on count data. We build a *DESeqDataSet*
and then specify the experimental design using a `~` and the variables
that we expect to produce differences in the counts. (These variables 
are used to assess how much technical variability is in the data, but
not used in the transformation function itself.)

```{r}
library(DESeq2)
dds <- DESeqDataSet(rse, ~condition + treatment)
```

We use this function, which implements a *variance stabilizing 
transformation* (more on this next time):

```{r}
vsd <- vst(dds, blind=FALSE)
```

As we did with microarray data, we calculate the variance 
across all samples (on the transformed data):

```{r}
library(matrixStats)
rv <- rowVars(assay(vsd))
```

Finally, before plotting a heatmap, we extract the covariates that 
we want to annotated the top of the plot.

```{r}
anno.col <- as.data.frame(colData(vsd)[,c("condition","treatment")])
anno.col
```

This code pull out the top of the transformed data by variance,
and adds an annotation to the top of the plot. By default the
rows and columns will be clustered by Euclidean distance. 
See `?pheatmap` for more details on this function (it's a very
detailed manual page).

```{r}
library(pheatmap)
pheatmap(assay(vsd)[head(order(rv, decreasing=TRUE),100),],
         annotation_col=anno.col,
         show_rownames=FALSE, show_colnames=FALSE)
```
